lib/puppet/provider/network_vlan/nxapi.rb

```diff
@@ -0,0 +1,55 @@
+$LOAD_PATH.unshift(File.join(File.dirname(__FILE__), '..', '..', '..'))
+require 'puppet/type/cisco_vlan'
+require 'puppet/provider/cisco_vlan/nxapi'
+
+Puppet::Type.type(:network_vlan).provide(:nxapi, :parent => Puppet::Type.type(:cisco_vlan).provider(:nxapi)) do
+  @doc = 'netdev_stdlib network_vlan'
+
+  has_features :activable, :describable
+
+  mk_resource_methods
+
+  def self.instances
+    vlans = []
+    Cisco::Vlan.vlans.each { |vlan_id, v|
+      vlan = {
+        id:         vlan_id,
+        name:       vlan_id,
+        vlan_name:  v.send(:vlan_name),
+        shutdown:   v.send(:shutdown),
+        ensure:     :present,
+      }
+      vlans << new(vlan)
+    }
+  end
+
+  def self.prefetch(resources)
```

Comment:
Is this needed or can you use the implementation of self.prefetch inherited from the parent class?